src/python/pants/backend/codegen/protobuf/python/rules.py

```diff
@@ -34,16 +42,7 @@ async def generate_python_from_protobuf(
     )
 
     output_dir = "_generated_files"
-    # TODO(#9650): replace this with a proper intrinsic to create empty directories.
-    create_output_dir_request = Get(
-        ProcessResult,
-        Process(
-            ("/bin/mkdir", output_dir),
-            description=f"Create the directory {output_dir}",
-            level=LogLevel.TRACE,
-            output_directories=(output_dir,),
-        ),
-    )
```

Comment:
Would it be useful to consider making a `Process` wrapper like `HackyLocalBinaryProcess` that uses the parent PATH to help make it easier to track these TODOs or at least help us maybe leave them until we have a process for discovering local binaries in a way we like?

Reply:
Or until we can cover that functionality with rust intrinsics or python rules.

Reply:
> a process for discovering local binaries in a way we like?

We do have this! `BinaryPaths`. See https://www.pantsbuild.org/v2.0/docs/rules-api-installing-tools#binarypaths-find-already-installed-binaries

Reply:
Ah! Just read https://www.pantsbuild.org/v2.0/docs/rules-api-installing-tools!

Reply:
I don't know how bazel does this (I think there are 3 different APIs for this) but I certainly know `make` doesn't do anything this principled about discovering local binaries and that could be a strong argument for people who don't like complexity because they don't like things breaking.

Reply:
I definitely remember hearing about this earlier, it's just really nice!